### ISSUE TITLE
Remove libglobus-gsi-credential1 from Debian repos

### ISSUE THREAD

**Issue by tpdownes** [cvmfs-contrib/cvmfs-contrib.github.io#2]
Thanks to good work by @ellert, Debian has updated the `globus-gsi-credential` source package with the fix necessary for OpenSSL 1.1 compatibility and has included it in [stretch-proposed-updates](https://release.debian.org/proposed-updates/stable.html). I've created a filtered mirror at
```
deb http://omen.phys.uwm.edu/software/globus stretch main
```
Weird thing. I can no longer reproduce the [original issue](https://github.com/globus/globus-toolkit/issues/115) with the stock debian package (`libglobus-gsi-credential1:amd64==7.11-1`). That is to say, I can run `grid-cert-diagnostics -c /tmp/x509up_p1788.filebpo7og.1` and get

```
Checking Certificate
====================
Checking cert at /tmp/x509up_p1788.filebpo7og.1... ok
Checking Certificate Subject... "/DC=org/DC=cilogon/C=US/O=LIGO/CN=Thomas Downes thomas.downes@ligo.org/CN=1627320063"
Checking cert... ok
Checking that certificate contains an RSA key... ok
Checking certificate trust chain... ok
```

And I can access `/cvmfs/ligo.osgstorage.org` using `cvmfs-x509-helper`.

It is possible that there was an update somewhere else (`libssl1.1`/`cvmfs-x509-helper`) that resolved it for us there? We weren't making this up! I've done the sanity check of destroying my proxy and trying again.

The behavior is unaffected if I upgrade to `7.11-1+deb9u1` or `7.11-1.cvmfs1u1`. So, that's good!

So, long story short, I can access `/cvmfs/ligo.osgstorage.org` with the original Debian.org package and with the proposed update. Therefore, I think the cvmfs-contrib package can be removed from the Deb8/9 cvmfs-contrib repos. It can be done on Debian 8 immediately. In principle, it could also be done immediately for Debian 9, though I'd appreciate someone actually checking my work with `libglobus-gsi-credential1:amd64==7.11-1`.

Regardless, it is important that that the cvmfs-contrib package get removed because `7.11-1.cvmfs1u1` wins the version #  over `7.11-1+deb9u1`. It should probably be removed from some/all of the Ubuntu repos, though I'm not certain which.

**Comment by tpdownes** [comment]
I've confirmed that I'm using ` /usr/bin/grid-cert-diagnostics` from `globus-proxy-utils==6.19-1`

**Comment by DrDaveD** [comment]
Shouldn't we wait until the package makes it from stretch-proposed-updates to stretch-updates?

I don't know about updates to libssl1.1, but cvmfs-x509-helper was updated recently to version 1.1.1.  It only [closed leaking file descriptors](https://github.com/cvmfs-contrib/cvmfs-x509-helper/pull/11) and that shouldn't have made a difference.

**Comment by tpdownes** [comment]
@DrDaveD It should definitely be gone from Deb 8.

Deb 9, arguably it can be removed because of the apparent change in behavior of `libglobus-gsi-credential1:amd64==7.11-1` (I even tried rebooting with this package installed). It should either be removed from Deb9 now or soon after promotion, so that the copy in `stretch-updates` would actually be used by package managers. I wouldn't remove it from Debian 9 without confirming my observation.

I can't speak to the behavior on various Ubuntu releases. I imagine it tracks Debian in a reasonably predictable way (i.e. only potentially needed when OpenSSL 1.1 is used and libglobus below < 7.14).

**Comment by DrDaveD** [comment]
It never was built for Debian 8.  I'll leave it in Debian 9 until it is in stretch-updates, which should be soon.  It is built for Ubuntu 17.04 and 18.04 and I don't know what the status on them is for the official patch.  I suppose I could try starting a docker container with them to find out, but I won't bother with that until some time after it is in stretch-updates.

**Comment by tpdownes** [comment]
It has always been available from your Deb8 repo and it remains so. On a Debian 8 machine w/o `cvmfs-contrib` configured:
```
root@pcdev1:~# apt-cache policy libglobus-gsi-credential1
libglobus-gsi-credential1:
  Installed: 7.7-3
  Candidate: 7.7-3
  Version table:
 *** 7.7-3 0
        500 http://omen.phys.uwm.edu/debian/ jessie/main amd64 Packages
        100 /var/lib/dpkg/status
```
Then configure it:
```
root@pcdev1:~# wget https://ecsft.cern.ch/dist/cvmfs/cvmfs-contrib-release/cvmfs-contrib-release-latest_all.deb
--2018-10-12 14:01:53--  https://ecsft.cern.ch/dist/cvmfs/cvmfs-contrib-release/cvmfs-contrib-release-latest_all.deb
Resolving ecsft.cern.ch (ecsft.cern.ch)... 188.184.161.78
Connecting to ecsft.cern.ch (ecsft.cern.ch)|188.184.161.78|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4316 (4.2K)
Saving to: ‘cvmfs-contrib-release-latest_all.deb’

cvmfs-contrib-release-latest_all.deb       100%[==========================================================================================>]   4.21K  --.-KB/s   in 0s

2018-10-12 14:01:54 (44.4 MB/s) - ‘cvmfs-contrib-release-latest_all.deb’ saved [4316/4316]

root@pcdev1:~# sudo dpkg -i cvmfs-contrib-release-latest_all.deb
Selecting previously unselected package cvmfs-contrib-release.
(Reading database ... 300497 files and directories currently installed.)
Preparing to unpack cvmfs-contrib-release-latest_all.deb ...
Unpacking cvmfs-contrib-release (1.4.1) ...
Setting up cvmfs-contrib-release (1.4.1) ...
root@pcdev1:~# rm -f cvmfs-contrib-release-latest_all.deb
root@pcdev1:~# sudo apt-get update
Hit http://omen.phys.uwm.edu jessie-backports InRelease
Hit http://omen.phys.uwm.edu jessie/updates InRelease
Hit http://omen.phys.uwm.edu jessie-updates InRelease
Ign http://omen.phys.uwm.edu jessie InRelease
Hit http://omen.phys.uwm.edu jessie InRelease
Hit http://omen.phys.uwm.edu jessie InRelease
Hit http://omen.phys.uwm.edu jessie InRelease
Hit http://omen.phys.uwm.edu jessie InRelease
Ign http://omen.phys.uwm.edu jessie InRelease
Hit http://omen.phys.uwm.edu jessie InRelease
Ign http://omen.phys.uwm.edu jessie InRelease
Hit http://omen.phys.uwm.edu jessie InRelease
Hit http://omen.phys.uwm.edu jessie Release.gpg
Hit http://omen.phys.uwm.edu jessie Release.gpg
Hit http://omen.phys.uwm.edu jessie Release.gpg
Hit http://omen.phys.uwm.edu jessie Release
Hit http://omen.phys.uwm.edu jessie Release
Hit http://omen.phys.uwm.edu jessie Release
Ign http://downloadcontent.opensuse.org ./ InRelease
Get:1 http://omen.phys.uwm.edu jessie-backports/main amd64 Packages/DiffIndex [27.8 kB]
Get:2 http://downloadcontent.opensuse.org ./ Release.gpg [481 B]
Ign http://cvmrepo.web.cern.ch jessie-prod InRelease
Get:3 http://omen.phys.uwm.edu jessie-backports/contrib amd64 Packages/DiffIndex [26.4 kB]
Get:4 http://omen.phys.uwm.edu jessie-backports/contrib Translation-en/DiffIndex [7,960 B]
Get:5 http://omen.phys.uwm.edu jessie-backports/main Translation-en/DiffIndex [27.8 kB]
Hit http://omen.phys.uwm.edu jessie/updates/main amd64 Packages
Hit http://omen.phys.uwm.edu jessie/updates/main Translation-en
Get:6 http://omen.phys.uwm.edu jessie-updates/main amd64 Packages/DiffIndex [11.8 kB]
Get:7 http://omen.phys.uwm.edu jessie-updates/main Translation-en/DiffIndex [3,688 B]
Get:8 http://downloadcontent.opensuse.org ./ Release [1,018 B]
Hit https://packagecloud.io jessie InRelease
Hit http://omen.phys.uwm.edu jessie/main amd64 Packages
Ign http://omen.phys.uwm.edu jessie/main Translation-en_US
Ign http://omen.phys.uwm.edu jessie/main Translation-en
Hit http://cvmrepo.web.cern.ch jessie-prod Release.gpg
Hit http://omen.phys.uwm.edu jessie/main amd64 Packages
Ign http://omen.phys.uwm.edu jessie/main Translation-en_US
Ign http://omen.phys.uwm.edu jessie/main Translation-en
Hit http://omen.phys.uwm.edu jessie/main amd64 Packages
Ign http://omen.phys.uwm.edu jessie/main Translation-en_US
Ign http://omen.phys.uwm.edu jessie/main Translation-en
Hit http://omen.phys.uwm.edu jessie/development amd64 Packages
Ign http://omen.phys.uwm.edu jessie/development Translation-en_US
Ign http://omen.phys.uwm.edu jessie/development Translation-en
Hit http://omen.phys.uwm.edu jessie/contrib amd64 Packages
Hit http://cvmrepo.web.cern.ch jessie-prod Release
Ign http://omen.phys.uwm.edu jessie/contrib Translation-en_US
Ign http://omen.phys.uwm.edu jessie/contrib Translation-en
Hit http://omen.phys.uwm.edu jessie/contrib amd64 Packages
Hit http://omen.phys.uwm.edu jessie/main amd64 Packages
Hit http://omen.phys.uwm.edu jessie/contrib amd64 Packages
Hit http://omen.phys.uwm.edu jessie/non-free amd64 Packages
Hit http://omen.phys.uwm.edu jessie/contrib Translation-en
Hit http://omen.phys.uwm.edu jessie/main Translation-en
Hit http://omen.phys.uwm.edu jessie/non-free Translation-en
Ign http://omen.phys.uwm.edu jessie/contrib Translation-en_US
Hit http://omen.phys.uwm.edu jessie/main amd64 Packages
Ign http://omen.phys.uwm.edu jessie/contrib Translation-en
Hit http://omen.phys.uwm.edu jessie/puppet5 amd64 Packages
Ign http://omen.phys.uwm.edu jessie/main Translation-en_US
Ign http://omen.phys.uwm.edu jessie/main Translation-en
Ign http://omen.phys.uwm.edu jessie/puppet5 Translation-en_US
Ign http://omen.phys.uwm.edu jessie/puppet5 Translation-en
Hit https://packagecloud.io jessie/main amd64 Packages
Get:9 http://downloadcontent.opensuse.org ./ Packages [2,467 B]
Get:10 https://packagecloud.io jessie/main Translation-en_US [162 B]
Get:11 https://packagecloud.io jessie/main Translation-en [162 B]
Hit http://cvmrepo.web.cern.ch jessie-prod/main amd64 Packages
Get:12 https://packagecloud.io jessie/main Translation-en_US [162 B]
Get:13 https://packagecloud.io jessie/main Translation-en [162 B]
Get:14 https://packagecloud.io jessie/main Translation-en_US [162 B]
Get:15 https://packagecloud.io jessie/main Translation-en [162 B]
Get:16 https://packagecloud.io jessie/main Translation-en_US [162 B]
Get:17 https://packagecloud.io jessie/main Translation-en [162 B]
Get:18 https://packagecloud.io jessie/main Translation-en_US [162 B]
Ign https://packagecloud.io jessie/main Translation-en_US
Get:19 https://packagecloud.io jessie/main Translation-en [162 B]
Ign https://packagecloud.io jessie/main Translation-en
Ign http://downloadcontent.opensuse.org ./ Translation-en_US
Ign http://downloadcontent.opensuse.org ./ Translation-en
Ign http://cvmrepo.web.cern.ch jessie-prod/main Translation-en_US
Ign http://cvmrepo.web.cern.ch jessie-prod/main Translation-en
Fetched 109 kB in 4s (23.6 kB/s)
Reading package lists... Done
root@pcdev1:~# apt-cache policy libglobus-gsi-credential1
libglobus-gsi-credential1:
  Installed: 7.7-3
  Candidate: 7.11-1.cvmfs1u1
  Version table:
     7.11-1.cvmfs1u1 0
        500 http://downloadcontent.opensuse.org/repositories/home:/cvmfs:/contrib/Debian_8.0/ ./ Packages
 *** 7.7-3 0
        500 http://omen.phys.uwm.edu/debian/ jessie/main amd64 Packages
        100 /var/lib/dpkg/status
```

**Comment by tpdownes** [comment]
P.S. Just to be really, really sure. I removed my `apt.conf.d` rule that uses a local forward proxy.
P.P.S. Whether one uses `download` or `downloadcontent` does not appear to affect matters.
P.P.P.S. Outward examination reveals a difference between the public repo and the drdaved repo.

**Comment by DrDaveD** [comment]
Ok, you're right, it was enabled for Debian 8 at one time, but it has been disabled for a long time and for some reason it hasn't removed the binaries.  I now found that if I click on the Debian 8 build of the package that the binaries were shown in a build directory and I clicked on the button to delete them.  However they're still not gone from the repository.  I think I need to figure out a way to rebuild the repository; maybe rebuilding another package will do it, I'm not sure.

**Comment by DrDaveD** [comment]
I rebuilt cvmfs-x509-helper for Debian 8 but it didn't help, the extra package did not go away.

**Comment by DrDaveD** [comment]
Ok it's gone now.  I found a slight difference in the configuration between contrib-testing and contrib, and making them the same made the disabled package go away.  The difference was enabling Debian 8 for "Publish flag", "Debuginfo flag", and "Use for Build flag", while leaving it disabled for "Build flag"; previously they were all disabled.  I don't know which of the flags made the difference.

**Comment by tpdownes** [comment]
Debian 9 Update 9.6 is scheduled for 10 November. It will include the relevant backport.

**Comment by tpdownes** [comment]
I have tested the new package `libglobus-gsi-credential1:amd64=7.11-1+deb9u1` and found it to be successful. Unfortunately it loses the version # battle with your repo. I can force an upgrade by de-prioritizing your repo relative to debian.org, but a removal would also be helpful.

```
libglobus-gsi-credential1:
  Installed: 7.11-1.cvmfs1u1
  Candidate: 7.11-1.cvmfs1u1
  Version table:
 *** 7.11-1.cvmfs1u1 500
        500 http://download.opensuse.org/repositories/home:/cvmfs:/contrib/Debian_9.0 ./ Packages
        100 /var/lib/dpkg/status
     7.11-1+deb9u1 500
        500 http://omen.phys.uwm.edu/debian stretch/main amd64 Packages
```

**Comment by DrDaveD** [comment]
I got it removed from contrib-testing, but only by deleting the entire package.  Nothing seems to happen when I just disable the Debian_9.0 repository.  Is it safe to delete it from Ubuntu 17.04 and 18.04 too now?

**Comment by DrDaveD** [comment]
Ok I was able to remove it by explicitly deleting the build products as I talked about on October 12.  So I'll leave Ubuntu for now but I would like to know when it is time to remove it from there too.

**Comment by tpdownes** [comment]
(I realize this is a superset of your list, but calling it out explicitly)

* Ubuntu 14.04 does not have OpenSSL 1.1
* Ubuntu 16.04 does not have OpenSSL 1.1
* Ubuntu 17.04 is EOL and did not have OpenSSL 1.1
* Ubuntu 17.10 is EOL and did not have OpenSSL 1.1
* Ubuntu 18.04 has OpenSSL 1.1 and `libglobus-gsi-credential1==7.14-1build1` which [contains the fix in question](http://changelogs.ubuntu.com/changelogs/pool/universe/g/globus-gsi-credential/globus-gsi-credential_7.14-1build1/changelog)

I interpret this as meaning you can drop the building of this package on all Debian-derived platforms.

**Comment by DrDaveD** [comment]
Ok it is now removed from all platforms.
